cmake/zstd.cmake

```diff
@@ -41,7 +41,8 @@ MACRO (MYSQL_USE_BUNDLED_ZSTD)
   SET(BUILD_BUNDLED_ZSTD 1)
   SET(ZSTD_LIBRARY zstd CACHE INTERNAL "Bundled zlib library")
   CHECK_ZSTD_VERSION(${CMAKE_SOURCE_DIR}/extra/zstd/lib)
-  INCLUDE_DIRECTORIES(BEFORE SYSTEM ${CMAKE_SOURCE_DIR}/extra/zstd/lib)
+  INCLUDE_DIRECTORIES(BEFORE SYSTEM ${CMAKE_SOURCE_DIR}/extra/zstd/lib
+                                    ${CMAKE_SOURCE_DIR}/extra/zstd/lib/dictBuilder)
```

Comment:
What were these fixes for?

Reply:
It fixes an issue with `-DWITH_ZSTD=bundled`:
```
In file included from /data/mysql-server/fb-8.0.17/rocksdb/options/cf_options.h:14,
                 from /data/mysql-server/fb-8.0.17/rocksdb/table/format.h:22,
                 from /data/mysql-server/fb-8.0.17/rocksdb/table/internal_iterator.h:14,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/pinned_iterators_manager.h:12,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/range_tombstone_fragmenter.h:15,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/memtable.h:19,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/memtable_list.h:17,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/column_family.h:17,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/db_impl/db_impl.h:22,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/arena_wrapped_db_iter.h:13,
                 from /data/mysql-server/fb-8.0.17/rocksdb/db/arena_wrapped_db_iter.cc:10:
/data/mysql-server/fb-8.0.17/rocksdb/util/compression.h:50:10: fatal error: zdict.h: No such file or directory
   50 | #include <zdict.h>
```